mediapipe/tasks/ios/vision/core/sources/MPPMask.mm

```diff
@@ -0,0 +1,157 @@
+// Copyright 2023 The MediaPipe Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "mediapipe/tasks/ios/vision/core/sources/MPPMask.h"
+#import "mediapipe/tasks/ios/common/sources/MPPCommon.h"
+#import "mediapipe/tasks/ios/common/utils/sources/MPPCommonUtils.h"
+
+namespace {
+template <typename T>
+T *allocateDataPtr(std::unique_ptr<T[]> &data, size_t length) {
+  data = std::unique_ptr<T[]>(new T[length]);
```

Comment:
Only uninitialised `unique_ptr`  instance variables are passed by reference to this call from within this file. So no values are discarded. They are initialised from within the function and the changes are visible outside the function scope too due to the pass by reference. Also unique_ptr.get() returns a pointer, which would be valid for the scope of the ptr.
Just saves a couple of lines of code  and also the need to specify the type for alloc and copy. 

Technically, if the passed in ptr is already initialised, it would discard the existing value.  To make it fool proof for initialised ptrs, we need to do a null check which is unnecessary here. Hence dropped it.

Nevertheless, I have added these lines in place now to avoid any further pushback during the ObjC review.